app/main/reducers/browserWindows-reducers.js

```diff
@@ -0,0 +1,47 @@
+/* @flow */
+
+/*
+ Copyright 2016 Mozilla
+
+ Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ this file except in compliance with the License. You may obtain a copy of the
+ License at http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software distributed
+ under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ specific language governing permissions and limitations under the License.
+ */
+
+import * as profileActionTypes from '../constants/profile-action-types';
+
+import Immutable from 'immutable';
+
+import electron from 'electron';
+
+/**
+ * Map `BrowserWindow.id` to BrowserWindow instances.
+ *
+ * We diff the current and previous state.  However, after a BrowserWindow is closed, its id
+ * property is no longer available.  (The error will say "Object has been destroyed".)  Therefore,
+ * to allow to compare to a previous state with a BrowserWindow that is no longer alive, we key
+ * by the BrowserWindow.id and take care to not fetch the id of closed windows.
+ */
+type State = Immutable.Map<number, electron.BrowserWindow>;
+
+export default function browserWindowsReducer(state: State = new Immutable.Map(),
+                                              action: Object): State {
+  console.log(`reducing ${JSON.stringify(action)}`);
```

Comment:
Is this intentionally logging for all actions (even those that aren't handled by this reducer)?  It's actually useful, just was surprised when running it to see logs for all the actions, as it might get noisy.